src/jit/lsra.cpp

```diff
@@ -3644,9 +3660,12 @@ void LinearScan::buildRefPositionsForNode(GenTree*                  tree,
         if (isCandidateLocalRef(tree))
         {
             // We always push the tracked lclVar intervals
-            varDefInterval = getIntervalForLocalVar(tree->gtLclVarCommon.gtLclNum);
-            defRefType     = refTypeForLocalRefNode(tree);
-            defNode        = tree;
+            LclVarDsc* varDsc = &compiler->lvaTable[tree->gtLclVarCommon.gtLclNum];
+            assert(varDsc->lvTracked);
+            unsigned varIndex = varDsc->lvVarIndex;
+            varDefInterval    = getIntervalForLocalVar(varIndex);
+            defRefType        = refTypeForLocalRefNode(tree);
```

Comment:
It turns out varIndex is used elsewhere in both cases here, but there was another case where it wasn't, so I created `getIntervalForLocalVarNode` as suggested.